rllib/models/tf/attention_net.py

```diff
@@ -445,33 +452,48 @@ def forward(self, input_dict: Dict[str, TensorType],
 
         # Concat. prev-action/reward if required.
         prev_a_r = []
+        # Prev actions: Try flattening into 1D array (one-hot discrete
+        # components). This only properly works with complex action spaces
+        # iff action flattening has been disabled.
         if self.use_n_prev_actions:
-            if isinstance(self.action_space, Discrete):
-                for i in range(self.use_n_prev_actions):
-                    prev_a_r.append(
-                        one_hot(input_dict[SampleBatch.PREV_ACTIONS][:, i],
-                                self.action_space))
-            elif isinstance(self.action_space, MultiDiscrete):
-                for i in range(0, self.use_n_prev_actions,
-                               self.action_space.shape[0]):
-                    prev_a_r.append(
-                        one_hot(
-                            tf.cast(
-                                input_dict[SampleBatch.PREV_ACTIONS]
-                                [:, i:i + self.action_space.shape[0]],
-                                tf.float32), self.action_space))
+            prev_n_actions = input_dict[SampleBatch.PREV_ACTIONS]
+            if self.model_config["_disable_action_flattening"]:
+                # Merge prev n actions into flat tensor.
+                flat = flatten_inputs_to_1d_tensor(
+                    prev_n_actions,
+                    spaces_struct=self.action_space_struct,
+                    time_axis=True,
+                )
+                # Fold time-axis into flattened data.
+                flat = tf.reshape(flat, [tf.shape(flat)[0], -1])
+                prev_a_r.append(flat)
             else:
-                prev_a_r.append(
-                    tf.reshape(
-                        tf.cast(input_dict[SampleBatch.PREV_ACTIONS],
-                                tf.float32),
-                        [-1, self.use_n_prev_actions * self.action_dim]))
+                if isinstance(self.action_space, Discrete):
```

Comment:
can you say a few words about this branch? above you have a comment saying "this only properly works ... iff action flattening is disabled".
but this branch is for the case when it's not disabled, so what are we doing here, and what should users expect?
thanks.

Reply:
Yeah, agree, the comments here are not clear enough. I'll improve these. ...

The point is:
- If action flattening is off (actions come into the model as they were previously sent to the env, either as complex actions or simple discrete, boxes, etc.., doesn't matter), then we can push them through the new flatten util.
- If action flattening is on (soon to be deprecated), we'll behave as-is.